### ISSUE TITLE
Supplemental Claims Error Handling from Benefits Team 1

### ISSUE THREAD

**Issue by eileen-coforma** [department-of-veterans-affairs/va.gov-team#50052]
## What does your team need support for? Check all that apply.
- [ ] Launching one or more new unauthenticated (static) VA.gov pages
- [ ] Making revisions to one or more existing unauthenticated (static) VA.gov page
- [x] Launching a new online form, app, or tool on VA.gov
- [ ] Making revisions to an existing online form, app, or tool on VA.gov
- [ ] Something else (please add details in the next question)

## What do you need help with?
  - [x] Content support
  - [ ] IA support
  
  
*Please be specific about the content or IA support you need and provide any additional documents and/or links that will help us help you. If we need to add a react widget to a page, be sure to provide the code here.*

## Will this new product be released incrementally (for instance 25% of users initially)? 
  - [x] Yes
  - [ ] No
 
**Note:** If you checked yes, we'll reach out to discuss details about the content in the react widget.
  
## When do you expect to launch your product to 100% of users?

December
  
*Please provide an estimated date so our team can create other relevant tickets.*
  
## Supporting artifacts
*Please provide supporting artifacts as available.*
- [Supp Claims Content Source of Truth](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/decision-reviews/Supplemental-Claims/design/sourceofcontenttruth.md#error-messaging)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/49541
- [Screens](https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/p/C22CAF85-3AE8-4837-9C8F-0C7145740CC1/canvas)
- In the comments below

## Will this work be going through the Collaboration Cycle?
- [ ] Yes 
- [ ] No

## When does this work need to be done?
- Estimated launch date:
- Estimated staging review date:
- Content and IA work needed by: 

## Do you plan to bring this to an upcoming content office hours session?
- [x] Yes, benefit content office hours (Thursdays, 3:00 p.m. to 3:45 p.m. ET)
- [ ] Yes, health  content office hours (Thursdays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] Yes, unauth office hours (Mondays, 11:00 a.m. to 11:30 a.m. ET)
- [ ] No, but I'd like to schedule time to talk about this request
- [ ] No, let's work asynchronously and meet if needed

**Note:** If we think this work would benefit from a collaborative session with you, we may ask you to bring it to office hours or set up a separate time to meet.

## About your team

- Team name: 
-	OCTO-DE product owner: Matt Self
-	Product manager: @saderagsdale Sade Ragsdale
-	Designer: @eileen-coforma Eileen Chi
-	FE engineer: @Mottie Robin Garrison
-	Product/team Slack channel: Benefits-team-1

## Next steps

**Once you’ve submitted this ticket, please post a link to this issue in the [#sitewide-content-ia](https://dsva.slack.com/channels/sitewide-content-ia) Slack channel and tag Randi Hecht.**

If you also need engineering support from the Public Websites team, fill out their [intake request form](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=Public+Websites%2C+Sitewide+content&labels=vsa-public-websites%2C+vsa%2C+vsa-public-websites-intake%2C+sitewide-content%2C+needs-grooming&template=public-websites-intake.md&title=%3CType+of+Request%3E+from+%3CTeam%3E).
  
If you need a page/URL redirected, a URL changed or a vanity URL set up, please submit a [Redirect, URL change, or vanity URL request](https://github.com/department-of-veterans-affairs/va.gov-team/issues/new?assignees=mnorthuis&labels=ia&template=redirect-request.md&title=Redirect+Request)

**Comment by eileen-coforma** [comment]
**Major errors:**

- Error handling for PDF upload errors (be sure we help them recover): The PDF you uploaded is too large; break down the file into individual scans or use a PDF reader to save an optimized version. 
- Error handling for not adding an issue: You must select at least one issue, so we can process your request
- Error handling for not adding any evidence : You must select at least one type of new evidence, so we can process your request
- Error handling for not authorizing release of medical records : You must authorize the release of medical records in order for us to get records directly from your doctor. Otherwise, you will need to upload your private medical records.

**Inline errors we have yet to work on:**

- missingIssue: ‘You must add the name of an issue',
- uniqueIssue: ‘You must enter a unique condition name',
- maxLength: `You can enter a maximum of 140 characters`,

- invalidDate: ‘You must provide a valid date',
- missingDecisionDate: ‘You must enter a decision date',
- invalidDateRange: (min, max) => `You must enter a year between ${min} and ${max}`,
- pastDate: ‘You must add a past decision date',
- newerDate: ‘You must add an issue with a decision date less than a year old',

**Inline errors we've already worked on:**

- invalidDate: You must provide a date
- contestedIssue: The start date must be in the future
- endDateBeforeStart: The end date must be in the future 
- endDateBeforeStart: The end date must be after the start date
- contestedIssue: You must select an eligible issue

- evidenceMissing: You must include at least one type of supporting evidence
- evidenceLocationMissing: You must add a treatment location
- evidenceLocationMaxLength: You can enter a maximum of 255 characters
- evidenceIssuesMissing: You must select one or more conditions (double-check)

- missingEmail: You must provide an email address
- missingPhone: You must provide a home or mobile phone number
- missingAddress: You must provide an address
- missingPrimary: You must choose a primary phone number

**Comment by eileen-coforma** [comment]
Progress and edits on this is being made on the [Content Source of Truth](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/decision-reviews/Supplemental-Claims/design/sourceofcontenttruth.md)

**Comment by bethpottsVADEPO** [comment]
@eileen-coforma I'm working on updates in the source of truth, but need the context of the flow that we were looking at together. Can you drop that link here? Sorry, I think you may have already passed that along somewhere, but I can't find it!

Will return to this work in the morning. I know we need to get this done.

**Comment by eileen-coforma** [comment]
@bethpottsVADEPO Apologies, I didn't attach the screens on the ticket. These should link to the Issues and Evidence flows that would be relevant for the inline errors.

Issues: https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/p/C22CAF85-3AE8-4837-9C8F-0C7145740CC1/canvas
Decisions: Part of issues

VA Evidence and Private Evidence: https://www.sketch.com/s/d2416db4-9a4f-4919-abe4-20ba4bdcfd89/p/274ABD98-987B-4C4D-BB93-8FCE3DD6C3DE/canvas

**Comment by bethpottsVADEPO** [comment]
@eileen-coforma Quick question: Will the 5 error alerts in the source of truth doc use this alert component:

![image](https://user-images.githubusercontent.com/66026798/205686399-83501edb-acef-4a8a-ae34-00b1c2a11414.png)

**Comment by bethpottsVADEPO** [comment]
@eileen-coforma I updated the source of truth doc. @Mottie Sorry that I have a second round of edits for some of these that you've already had to update once. I realized some of the headers felt a little accusatory, which we try to stay away from (the approach is to sort of co-own with the user whatever the problem is--even if it's their misstep). Let me know if there are any questions, and I think I will need to revisit a small handful of inline errors that I had questions about. Those are inline in the source of truth doc.
